processor/spanmetricsprocessor/processor.go

```diff
@@ -59,10 +62,18 @@ type exemplarData struct {
 	value   float64
 }
 
+// metricKey is used to carry the stringified metric attributes
 type metricKey string
 
+// resourceKey is used to carry the stringified resource attributes
+type resourceKey string
+
 type processorImp struct {
-	lock   sync.RWMutex
+	// Based on OTEL Collector design, each pipeline will always get its own instance of the processor. ConsumeTraces()
+	// of each instance should never be called concurrently in practice.
```

Comment:
> Based on OTEL Collector design, each pipeline will always get its own instance of the processor. 

This is true.

> ConsumeTraces() of each instance should never be called concurrently in practice.

This is not.  `ConsumeTraces()` will be invoked by the previous component in the pipeline.  That component may itself be handling multiple concurrent invocations of its own `ConsumeTraces()` method or it may be handling multiple concurrent network requests.

Reply:
Thanks for clarifying this. I'll update the comment.